impl/src/main/java/org/jboss/cdi/tck/tests/beanContainer/BeanContainerTest.java

```diff
@@ -161,4 +167,44 @@ public void testGetContexts() {
         assertTrue(builtInContextImpls.size() >= 1);
     }
 
+    @Test
+    @SpecAssertion(section = BM_PROXY_UNWRAP, id = "a")
+    public void testUnwrapClientProxyForNormalScope() {
+        ApplicationScopedBean applicationScoped = getContextualReference(ApplicationScopedBean.class);
+        assertNotSame(applicationScoped, getCurrentBeanContainer().unwrapClientProxy(applicationScoped));
+    }
+
+    @Test
+    @SpecAssertion(section = BM_PROXY_UNWRAP, id = "b")
+    public void testUnwrapClientProxyForPseudoScope() {
+        DependentBean dependent = getContextualReference(DependentBean.class);
+        assertSame(dependent, getCurrentBeanContainer().unwrapClientProxy(dependent));
+    }
+
+    @Test
+    @SpecAssertion(section = BM_PROXY_UNWRAP, id = "b")
+    public void testUnwrapClientProxyForNull() {
+        assertNull(getCurrentBeanContainer().unwrapClientProxy(null));
+    }
+
+    @Test
+    @SpecAssertion(section = BM_PROXY_UNWRAP, id = "c")
+    public void testUnwrapClientProxyForInactiveScope() {
+        Contexts<Context> ctx = getCurrentConfiguration().getContexts();
+        ctx.setInactive(ctx.getRequestContext());
+
+        RequestScopedBean requestScoped = getContextualReference(RequestScopedBean.class);
+        assertThrows(ContextNotActiveException.class, () -> {
+            getCurrentBeanContainer().unwrapClientProxy(requestScoped);
+        });
+    }
+
+    @Test
+    @SpecAssertion(section = BM_PROXY_UNWRAP, id = "c")
```

Comment:
IMO this should be also (or even instead?) linked to a section dealing with https://jakarta.ee/specifications/cdi/4.1/jakarta-cdi-spec-4.1#active_context

Reply:
No, the specification of `unwrapClientProxy()` directly says that the method rethrows ISE, so I believe the reference is right.

Reply:
Done.